homeassistant/components/sensibo/config_flow.py

```diff
@@ -38,32 +36,67 @@ async def async_validate_api(hass: HomeAssistant, api_key: str) -> bool:
 
     try:
         async with async_timeout.timeout(TIMEOUT):
-            if await client.async_get_devices():
-                return True
+            await client.async_get_devices()
+    except AuthenticationError as err:
+        LOGGER.error("Failed to authenticate to Sensibo servers %s", err)
+        return "invalid_auth"
     except (
         aiohttp.ClientConnectionError,
         asyncio.TimeoutError,
-        AuthenticationError,
         SensiboError,
     ) as err:
-        _LOGGER.error("Failed to get devices from Sensibo servers %s", err)
-    return False
+        LOGGER.error("Failed to get devices from Sensibo servers %s", err)
+        return "cannot_connect"
+    return "validated"
 
 
-class SensiboConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+class SensiboConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Sensibo integration."""
 
     VERSION = 1
 
-    async def async_step_import(self, config: dict) -> FlowResult:
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
         """Import a configuration from config.yaml."""
 
-        self.context.update(
-            {"title_placeholders": {"Sensibo": f"YAML import {DOMAIN}"}}
-        )
         return await self.async_step_user(user_input=config)
 
-    async def async_step_user(self, user_input=None) -> FlowResult:
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with Sensibo."""
+        errors: dict[str, str] = {}
+
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
```

Comment:
Store this and move to a `async_step_reauth_confirm` step since if they reenter this the context won't be set.